### DISCUSSION TITLE
Lint rules based on targeted support engine in package.json

### DISCUSSION THREAD

**Discussion by jimmywarting** [eslint/eslint#15113] [Ideas] [answered]
It's very annoying to have to configure each project that targets different node versions, and also knowing what you can/can't use like `?.fn()`

Like browserslist and babels dynamic preset-evt. I would like for you to check eg if package.json contains support for `"engines": { "node": "=>0.10" }` or something like that

if it still supports a very old NodeJS version then it should forbid stuff like `const/let`
likewise it should disable the rule "prefer let/const" rule

---

doe this would only apply if you use build-less setups where you don't use babel or something

**Comment by nzakas** [comment, marked as answer]
You may want to use this plugin:
https://github.com/mysticatea/eslint-plugin-node

**Reply by jimmywarting** [reply]
Cool, thanks
